`ERROR: AssertionError: package 'Polyhedra' not loaded (it is required for executing tohrep)` despite it being present

I'm loading LazySets and Polyhedra (in that order) in some of my code to compute the `LazySets.constraints_list` of a `VPolytope`. This used to work but now it errors with:

```julia
ERROR: AssertionError: package 'Polyhedra' not loaded (it is required for executing `tohrep`)
Stacktrace:
  [1] require(mod::Module, package::Symbol; fun_name::String, explanation::String)
    @ ReachabilityBase.Require ~/.julia/packages/ReachabilityBase/VQJKD/src/Require/Require.jl:47
  [2] #require#1
    @ ~/.julia/packages/ReachabilityBase/VQJKD/src/Require/Require.jl:41 [inlined]
  [3] tohrep(P::LazySets.VPolytope{Float64, Vector{Float64}}; backend::Polyhedra.DefaultLibrary{Float64})
    @ LazySets ~/.julia/packages/LazySets/OP742/src/Sets/VPolytope.jl:487
  [4] tohrep
    @ ~/.julia/packages/LazySets/OP742/src/Sets/VPolytope.jl:481 [inlined]
```

--- a/src/Approximations/symmetric_interval_hull.jl
+++ b/src/Approximations/symmetric_interval_hull.jl
@@ -119,7 +119,7 @@ function symmetric_interval_hull(lm::LinearMap{N, <:AbstractHyperrectangle}) whe
 end
 
 function symmetric_interval_hull(E::ExponentialMap{N, <:AbstractSingleton}) where {N}
-    require(:Expokit; fun_name="symmetric_interval_hull")
+    require(@__MODULE__, :Expokit; fun_name="symmetric_interval_hull")
 
     v = expmv(one(N), E.spmexp.M, element(E.X))
     c = zeros(N, dim(E))
@@ -128,7 +128,7 @@ function symmetric_interval_hull(E::ExponentialMap{N, <:AbstractSingleton}) wher
 end
 
 function symmetric_interval_hull(E::ExponentialMap{N, <:AbstractHyperrectangle}) where {N}
-    require(:Expokit; fun_name="symmetric_interval_hull")
+    require(@__MODULE__, :Expokit; fun_name="symmetric_interval_hull")
 
     H = set(E)
     n = dim(H)

--- a/src/Approximations/underapproximate.jl
+++ b/src/Approximations/underapproximate.jl
@@ -80,7 +80,7 @@ arXiv:1905.13246.
 """
 function underapproximate(X::ConvexSet{N}, ::Type{<:Hyperrectangle};
                           solver=nothing) where {N}
-    require(:Ipopt; fun_name="underapproximate")
+    require(@__MODULE__, :Ipopt; fun_name="underapproximate")
 
     solver = default_nln_solver(N)
     return _underapproximate_box(X, solver)

--- a/src/ConcreteOperations/cartesian_product.jl
+++ b/src/ConcreteOperations/cartesian_product.jl
@@ -84,7 +84,7 @@ For further information on the supported backends see
 [Polyhedra's documentation](https://juliapolyhedra.github.io/).
 """
 function cartesian_product(P1::VPolytope, P2::VPolytope; backend=nothing)
-    require(:Polyhedra; fun_name="cartesian_product")
+    require(@__MODULE__, :Polyhedra; fun_name="cartesian_product")
 
     return _cartesian_product_vrep(P1, P2, backend1=backend, backend2=backend)
 end
@@ -123,7 +123,7 @@ function _cartesian_product_hrep(X::S1, Y::S2) where {S1<:ConvexSet{N}, S2<:Conv
 end
 
 function _cartesian_product_hrep_polyhedra(P1::PT1, P2::PT2; backend1, backend2) where {PT1, PT2}
-    require(:Polyhedra; fun_name="`cartesian_product")
+    require(@__MODULE__, :Polyhedra; fun_name="`cartesian_product")
 
     if isnothing(backend1)
         backend1 = default_polyhedra_backend(P1)

--- a/src/ConcreteOperations/convex_hull.jl
+++ b/src/ConcreteOperations/convex_hull.jl
@@ -581,7 +581,7 @@ For further information on the supported backends see
 """
 function convex_hull(P1::HPoly, P2::HPoly;
                      backend=default_polyhedra_backend(P1))
-    require(:Polyhedra; fun_name="convex_hull")
+    require(@__MODULE__, :Polyhedra; fun_name="convex_hull")
     Pch = Polyhedra.convexhull(polyhedron(P1; backend=backend),
                                polyhedron(P2; backend=backend))
     removehredundancy!(Pch)

--- a/src/ConcreteOperations/minkowski_sum.jl
+++ b/src/ConcreteOperations/minkowski_sum.jl
@@ -180,8 +180,8 @@ end
 
 # common code before calling _minkowski_sum_hrep
 function _minkowski_sum_hrep_preprocess(P, Q, backend, algorithm, prune)
-    require(:Polyhedra; fun_name="minkowski_sum")
-    require(:CDDLib; fun_name="minkowski_sum")
+    require(@__MODULE__, :Polyhedra; fun_name="minkowski_sum")
+    require(@__MODULE__, :CDDLib; fun_name="minkowski_sum")
 
     A, b = tosimplehrep(P)
     C, d = tosimplehrep(Q)
@@ -472,7 +472,7 @@ function _minkowski_sum_vrep_nd(vlist1::Vector{VT}, vlist2::Vector{VT};
     end
     if apply_convex_hull
         if backend == nothing
-            require(:Polyhedra; fun_name="minkowski_sum")
+            require(@__MODULE__, :Polyhedra; fun_name="minkowski_sum")
             backend = default_polyhedra_backend_nd(N)
             solver = default_lp_solver_polyhedra(N)
         end

--- a/src/Interfaces/AbstractPolyhedron_functions.jl
+++ b/src/Interfaces/AbstractPolyhedron_functions.jl
@@ -25,7 +25,7 @@ end
 
 # Polyhedra backend (fallback method)
 function default_polyhedra_backend(P::ConvexSet{N}) where {N}
-    require(:Polyhedra; fun_name="default_polyhedra_backend")
+    require(@__MODULE__, :Polyhedra; fun_name="default_polyhedra_backend")
     error("no default backend for numeric type $N")
 end
 
@@ -33,7 +33,7 @@ end
 # NOTE: exists in parallel to `default_lp_solver` because we use different
 # interfaces (see #1493)
 function default_lp_solver_polyhedra(N, varargs...)
-    require(:Polyhedra; fun_name="default_lp_solver_polyhedra")
+    require(@__MODULE__, :Polyhedra; fun_name="default_lp_solver_polyhedra")
     error("no default solver for numeric type $N")
 end
 
@@ -379,9 +379,9 @@ function _check_algorithm_applies(M::AbstractMatrix{N}, P::AbstractPolyhedron{N}
 end
 
 function _get_elimination_instance(N, backend, elimination_method)
-    require(:Polyhedra; fun_name="linear_map with elimination")
+    require(@__MODULE__, :Polyhedra; fun_name="linear_map with elimination")
     if backend == nothing
-        require(:CDDLib; fun_name="linear_map with elimination")
+        require(@__MODULE__, :CDDLib; fun_name="linear_map with elimination")
         backend = default_cddlib_backend(N)
     end
     if elimination_method == nothing
@@ -679,7 +679,7 @@ function _linear_map_vrep(M::AbstractMatrix, P::AbstractPolyhedron,
         throw(ArgumentError("the linear map in vertex representation for an " *
                             "unbounded set is not defined"))
     end
-    require(:Polyhedra; fun_name="linear_map",
+    require(@__MODULE__, :Polyhedra; fun_name="linear_map",
             explanation="of a $(typeof(P)) by a non-invertible matrix")
     # since P is bounded, we pass an HPolytope and then convert it to vertex representation
 

--- a/src/Interfaces/AbstractPolytope.jl
+++ b/src/Interfaces/AbstractPolytope.jl
@@ -175,7 +175,7 @@ The volume of `P`.
 The volume is computed by the `Polyhedra` library.
 """
 function volume(P::AbstractPolytope; backend=nothing)
-    require(:Polyhedra; fun_name="volume")
+    require(@__MODULE__, :Polyhedra; fun_name="volume")
     if isnothing(backend)
         backend = default_polyhedra_backend(P)
     end

--- a/src/Interfaces/ConvexSet.jl
+++ b/src/Interfaces/ConvexSet.jl
@@ -1526,7 +1526,7 @@ function chebyshev_center_radius(P::ConvexSet{N};
                                  backend=default_polyhedra_backend(P),
                                  solver=default_lp_solver_polyhedra(N; presolve=true)
                                 ) where {N}
-    require(:Polyhedra; fun_name="chebyshev_center")
+    require(@__MODULE__, :Polyhedra; fun_name="chebyshev_center")
     # convert to HPolyhedron to ensure `polyhedron` is applicable (see #1505)
     Q = polyhedron(convert(HPolyhedron, P); backend=backend)
     c, r = Polyhedra.chebyshevcenter(Q, solver)

--- a/src/LazyOperations/Intersection.jl
+++ b/src/LazyOperations/Intersection.jl
@@ -273,7 +273,7 @@ function ρ_helper(d::AbstractVector{M},
         NN = promote_type(N, M)
         return invoke(ρ, Tuple{typeof(d), Intersection{NN}}, d, cap)
     elseif algorithm == "line_search"
-        require(:Optim; fun_name="ρ", explanation="(algorithm $algorithm)")
+        require(@__MODULE__, :Optim; fun_name="ρ", explanation="(algorithm $algorithm)")
         (s, _) = _line_search(d, X, H; kwargs...)
         return s
     elseif algorithm == "projection"

--- a/src/Plotting/mesh.jl
+++ b/src/Plotting/mesh.jl
@@ -118,8 +118,8 @@ julia> plot3d!(10. * rand(Hyperrectangle, dim=3), color=:red)
 function plot3d(S::ConvexSet; backend=default_polyhedra_backend(S),
                 alpha=1.0, color=:blue, colormap=:viridis, colorrange=nothing, interpolate=false,
                 linewidth=1, overdraw=false, shading=true, transparency=true, visible=true)
-    require(:Makie; fun_name="plot3d")
-    require(:Polyhedra; fun_name="plot3d")
+    require(@__MODULE__, :Makie; fun_name="plot3d")
+    require(@__MODULE__, :Polyhedra; fun_name="plot3d")
 
     if colorrange == nothing
         colorrange = Automatic()
@@ -149,8 +149,8 @@ See the documentation of `plot3d` for examples.
 function plot3d!(S::ConvexSet; backend=default_polyhedra_backend(S),
                 alpha=1.0, color=:blue, colormap=:viridis, colorrange=nothing, interpolate=false,
                 linewidth=1, overdraw=false, shading=true, transparency=true, visible=true)
-    require(:Makie; fun_name="plot3d!")
-    require(:Polyhedra; fun_name="plot3d!")
+    require(@__MODULE__, :Makie; fun_name="plot3d!")
+    require(@__MODULE__, :Polyhedra; fun_name="plot3d!")
 
     if colorrange == nothing
         colorrange = Automatic()

--- a/src/Sets/Ball2.jl
+++ b/src/Sets/Ball2.jl
@@ -324,7 +324,7 @@ See `_sample_unit_nball_muller!` for implementation details.
 function sample(B::Ball2{N}, nsamples::Int;
                 rng::AbstractRNG=GLOBAL_RNG,
                 seed::Union{Int, Nothing}=nothing) where {N}
-    require(:Distributions; fun_name="sample")
+    require(@__MODULE__, :Distributions; fun_name="sample")
     n = dim(B)
     D = Vector{Vector{N}}(undef, nsamples) # preallocate output
     _sample_unit_nball_muller!(D, n, nsamples, rng=rng, seed=seed)

--- a/src/Sets/HPolyhedron.jl
+++ b/src/Sets/HPolyhedron.jl
@@ -450,7 +450,7 @@ documentation](https://juliapolyhedra.github.io/).
 """
 function tovrep(P::HPoly;
                 backend=default_polyhedra_backend(P))
-    require(:Polyhedra; fun_name="tovrep")
+    require(@__MODULE__, :Polyhedra; fun_name="tovrep")
     P = polyhedron(P; backend=backend)
     return VPolytope(P)
 end
@@ -505,7 +505,7 @@ function isempty(P::HPoly{N},
         return witness ? (false, an_element(P)) : false
     end
     if use_polyhedra_interface
-        require(:Polyhedra; fun_name="isempty", explanation="with the active " *
+        require(@__MODULE__, :Polyhedra; fun_name="isempty", explanation="with the active " *
             "option `use_polyhedra_interface`")
 
         if backend == nothing

--- a/src/Sets/HPolytope.jl
+++ b/src/Sets/HPolytope.jl
@@ -105,7 +105,7 @@ function rand(::Type{HPolytope};
               rng::AbstractRNG=GLOBAL_RNG,
               seed::Union{Int, Nothing}=nothing,
               num_vertices::Int=-1)
-    require(:Polyhedra; fun_name="rand")
+    require(@__MODULE__, :Polyhedra; fun_name="rand")
     rng = reseed(rng, seed)
     vpolytope = rand(VPolytope; N=N, dim=dim, rng=rng, seed=seed,
                     num_vertices=num_vertices)
@@ -230,7 +230,7 @@ function vertices_list(P::HPolytope{N};
     if dim(P) == 2 && backend == nothing
         return vertices_list(convert(HPolygon, P, prune=prune))
     else
-        require(:Polyhedra; fun_name="vertices_list")
+        require(@__MODULE__, :Polyhedra; fun_name="vertices_list")
         if backend == nothing
             backend = default_polyhedra_backend(P)
         end

--- a/src/Sets/VPolytope.jl
+++ b/src/Sets/VPolytope.jl
@@ -437,7 +437,7 @@ Otherwise, the redundancy removal function of the polyhedral backend is used.
 function remove_redundant_vertices(P::VPolytope{N};
                                    backend=nothing,
                                    solver=nothing) where {N}
-    require(:Polyhedra; fun_name="remove_redundant_vertices")
+    require(@__MODULE__, :Polyhedra; fun_name="remove_redundant_vertices")
     if backend == nothing
         backend = default_polyhedra_backend(P)
     end
@@ -484,7 +484,7 @@ function tohrep(P::VPolytope{N};
     if isempty(vl)
         return EmptySet{N}(dim(P))
     end
-    require(:Polyhedra; fun_name="tohrep")
+    require(@__MODULE__, :Polyhedra; fun_name="tohrep")
     return convert(HPolytope, polyhedron(P; backend=backend))
 end
 

